app/controllers/chatrooms_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ChatroomsController < ApplicationController
+
+  def index
+    @chatroom = Chatroom.new
+    @chatrooms = Chatroom.all
+  end
+
+  def new
+    if request.referrer.split("/").last == "chatrooms"
+      flash[:notice] = nil
+    end
+    @chatroom = Chatroom.new
+  end
+
+  def edit
+    @chatroom = Chatroom.find_by(slug: params[:slug])
```

Comment:
Down the line, the way I can think of to make each of these unique is to automatically create a new chatroom for each confirmed walk. 

We can search by walk id or and say that each chat belongs to a walk and each walk has one chat.